File: server/app/schemas/report.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel

from app.models.enums import ReportCategory, ReportStatus, VarnaDistrict


class ReportBase(BaseModel):
    title: str
    description: str
    category: ReportCategory
    district: VarnaDistrict
    address: str
    latitude: float
    longitude: float


class ReportCreate(ReportBase):
    image_url: Optional[str] = None


class ReportUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    category: Optional[ReportCategory] = None
    district: Optional[VarnaDistrict] = None
    address: Optional[str] = None
    latitude: Optional[float] = None
    longitude: Optional[float] = None
    image_url: Optional[str] = None
    admin_note: Optional[str] = None
    status: Optional[ReportStatus] = None


class ReportRead(ReportBase):
    id: int
    status: ReportStatus
    image_url: Optional[str] = None
    admin_note: Optional[str] = None
    created_by_id: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

File: client/src/features/reports/types/report.ts
export type ReportStatus =
	| 'NEW'
	| 'UNDER_REVIEW'
	| 'IN_PROGRESS'
	| 'RESOLVED'
	| 'REJECTED'

export type ReportCategory =
	| 'ROAD'
	| 'PAVEMENT'
	| 'STREETLIGHT'
	| 'BUILDING'
	| 'WASTE'
	| 'PARK'
	| 'OTHER'

export type VarnaDistrict =
	| 'ASPARUHOVO'
	| 'PRIMORSKI'
	| 'ODESSOS'
	| 'MLADOST'
	| 'VLADISLAV_VARNENCHIK'

export type Report = {
	id: number
	title: string
	description: string
	category: ReportCategory
	district: VarnaDistrict
	address: string
	latitude: number
	longitude: number
	image_url?: string | null
	admin_note?: string | null
	created_by_id: number
	status: ReportStatus
	created_at: string
	updated_at: string
}

export type CreateReportPayload = {
	title: string
	description: string
	category: ReportCategory
	district: VarnaDistrict
	address: string
	latitude: number
	longitude: number
	image_url?: string | null
}

export type UpdateReportPayload = Partial<CreateReportPayload> & {
	admin_note?: string | null
	status?: ReportStatus
}
